chiron/integrators.py

```diff
@@ -194,20 +229,29 @@ def _wrap_and_rebuild_neighborlist(self, x: jnp.array, nbr_list: PairsBase):
             The coordinates of the particles.
         nbr_list: PairsBsse
             The neighborlist object.
+
+        Returns
+        -------
+        x: jnp.array
+            The wrapped coordinates.
+        nbr_list: PairsBase
+            The neighborlist object; this may or may not have been rebuilt.
         """
 
         x = nbr_list.space.wrap(x)
         # check if we need to rebuild the neighborlist after moving the particles
         if nbr_list.check(x):
```

Comment:
The way the algorithm is coded up now, to evaluate if we need to rebuild the list it just calculates the displacement from a reference configuration (with the minimum image convention applied). If the particles aren't wrapped before the evaluation, we would not need to apply the wrap function...but I'm not sure we can guarantee that particles in the neighborlist haven't been wrapped (I think adding a flag to the check function that toggles whether or not to wrap would solve this, with the default being true for "safety).

Reply:
I think I'm going to have to revisit this later after some more though. The neighbor and pairlist codes do not track the particle image; the way the displacement and wrapping functions in the OrthogonalPeriodicSpace class are implemented,  we don't need to actually know the "image" or do any corrections to get correct the correct displacement or wrapping). So in reality, we do not need to wrap at all and will still be able to get the correct displacement (I also assume we will not be simulating high temperature gases  where we would be concerned with coordinates growing really large if we don't periodically wrap).  

So I guess the question is: do we want to periodically wrap particles in the box? While we could use the nbr_list.check as the trigger to wrap particles, that only would apply to NeighborListNsqrd, as the `check` function in the brute force PairListNsqrd is basically always false (no rebuild)...only is true if we have a change in the number of particles.  So if we do want to wrap, it would be something we could do after a moveset completes, but again not sure if necessary.  

I feel as though we might want to provide the option to wrap the coordinates being written to the xtc file, as that is often helpful for visualization and certain types of data analysis, but obviously wrapping can be done via tools outside of the code. 

I think it might be best to just address this in a separate PR so I can do more checking to make sure I'm not totally missing something obvious that will break everything or introduce weird edge cases.